### DISCUSSION TITLE
Woocommerce Integration issues

### DISCUSSION THREAD

**Discussion by ImMrDC** [flyntwp/flynt#450] [Q&A] [answered]
I am trying to integrate Woocommerce into a Flynt website. I've gotten some of the way, but I am having issues. I have already spent months on the non-store aspect of this clients website, and understandably do not want to have start again with a non-flynt build because I can't get a simple Woocomerce store to run on Flynt/Timber.

For the purposes of debugging I've started a clean wp/flynt/woo install and used the sample product data. I have then followed this tutorial: https://timber.github.io/docs/guides/woocommerce/

The issue may be at this point mentioned in the tutorial: _"Note: Some users reported issues with the loop context even when using the timber_set_product() helper function. Turns out the default WooCommerce hooks interfere with the output of the aforementioned function."_

As a result, in the "tease-product.twig" - things like post.link and post.title are correct, but the actions (eg: woocommerce_after_shop_loop_item_title) seems to return the first loop items info. This can be fixed by creating custom thumbnail code with post.thumbnail and such, but this issue seems to continue on the single-product page.

**The main issue:**
On the single product page, if the product is a variable product, info directly on the single-product.twig page returns correctly, but any woocommerce actions return info as if the product is a simple product. This makes the product unpurchaseable, as you can't select a variation.

The variation add-to-cart form is complex, and can't be easily recreated like a thumbnail. It seems there is likely 1 thing interfering with the right info for getting to the actions. Unfortunately I'm not adept enough to figure it out.

Has anyone else come across hits? fixed it? Found a work around? If I can get the variations to work, this store will still be feasible. But it seems there may only be 1 thing standing in the way.

![image](https://user-images.githubusercontent.com/34407719/179726501-f58d9113-6f81-449a-8a66-f58352c31b91.png)
You can see above that it has the correct info - title, images, even the price is correct - however the price should be a range, and the add to cart should have variation selectors for the attributes.

**Comment by ImMrDC** [comment]
OK - small progress.

It seems If I modify one of the templates called by a woocommerce actions - eg: plugins/woocommerce/templates/single-product/price.php - And change `global $product;` to `global $product;
$product = wc_get_product( get_the_ID() );` - It works, and has a knock on effect in this case to show the correct add to cart form with the variations. Yay. 

But why? I don't want to have to modify every template to include this line. Why is `global $product;` not sufficient? Is there a way to modify how `global $product;` work so that it would cover all instances of it? I'm not even sure that makes sense.

**Comment by timohubois** [comment, marked as answer]
@ImMrDC in a [previous discussion](https://github.com/flyntwp/flynt/discussions/358#discussioncomment-1706418), there was a suggestion for an other guide: https://github.com/mindkomm/timber-integration-woocommerce

Hope this helps?
